pandas/core/dtypes/dtypes.py

```diff
@@ -999,7 +1000,7 @@ def __str__(self) -> str_type:
 
     @property
     def name(self) -> str_type:
-        return f"period[{self._freqstr}]"
+        return "".join(["period[", freq_to_period_freqstr(1, self._freqstr), "]"])
```

Comment:
I checked, it seems to be correct
```
>>> pd.PeriodDtype(freq="2M")
period[2M]
```

Reply:
I think you don't need this at all - you've already corrected `self._freqstr`, so this could just stay as
```
        return f"period[{self._freqstr}]"
```